frontend/src/lib/hooks/useMessagesFiltersFields.ts

```diff
@@ -0,0 +1,94 @@
+import { MessagesFilterKeysTypes } from 'lib/types';
+import { MessagesFilterKeys } from 'lib/constants';
+import { useLocalStorage } from 'lib/hooks/useLocalStorage';
+
+type MessagesFilterFieldsType = Pick<
+  Partial<{
+    [key in MessagesFilterKeysTypes]: string;
+  }>,
+  | 'mode'
+  | 'offset'
+  | 'timestamp'
+  | 'partitions'
+  | 'keySerde'
+  | 'valueSerde'
+  | 'stringFilter'
+>;
+
+export function useMessagesFiltersFields(topicName: string) {
+  const [messageFilters, setMessageFilters] = useLocalStorage<{
+    [topicName: string]: MessagesFilterFieldsType;
+  }>('message-filters-fields', {});
+
+  const removeMessagesFilterFields = () => {
+    setMessageFilters((prev) => {
+      const { [topicName]: topicFilters, ...rest } = prev || {};
+      return rest;
+    });
+  };
+
+  const removeMessagesFiltersField = (key: keyof MessagesFilterFieldsType) => {
```

Comment:
```suggestion
  const removeMessagesFiltersField = (key: MessagesFilterFieldsType) => {
```

because keys and values are the same we can use it like this

Reply:
but then typescript won't like it
![image](https://github.com/user-attachments/assets/bfc452a3-41df-4dcb-b907-23a7a0935950)